### PR TITLE
add reshape codec

### PR DESCRIPTION
**Pull request by konstibob**
Implemented a spec-compliant ReshapeCodec https://github.com/zarr-developers/zarr-extensions/pull/10 in zarr-java plus tests, all passing.

- Codec: array→array reshape preserving C-order ravel; validates all spec rules (input_dims, -1, product invariant, strictly-increasing dims, alignment). 
- Tests: round-trips (merge/split/flatten/singletons/-1/high-dim→1D), metadata resolution, ~11 invalid-config edge cases, and end-to-end read/write (incl. combined with transpose).

### CODE REVIEW

File: src/main/java/dev/zarr/zarrjava/v3/codec/core/ReshapeCodec.java [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+    @Override
+    public Array encode(Array chunkArray) throws ZarrException {
+        int[] inputShape = arrayMetadata.chunkShape;
+        if (!Arrays.equals(chunkArray.getShape(), inputShape)) {
+            throw new ZarrException(
+                    "reshape codec received an array of shape " + Arrays.toString(chunkArray.getShape())
+                            + " but expected the chunk shape " + Arrays.toString(inputShape) + ".");
+        }
+        int[] outputShape = resolveOutputShape(inputShape);
```
Comment (normanrz) [inline src/main/java/dev/zarr/zarrjava/v3/codec/core/ReshapeCodec.java]: The `outputChunkShape` is computed in every `encode` and `decode` call. It could be cached as a protected property of the class, because it is only changed when the `arrayMetadata` changes. You can override the `setCoreArrayMetadata` metadata, compute the `outputChunkShape` there, and use the property in `encode` and `decode`.

File: src/test/java/dev/zarr/zarrjava/ReshapeCodecTest.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        Assertions.assertArrayEquals(expectedOutputShape, encoded.getShape());
+        // The lexicographical (C-order) ravel is preserved: ravel(B) == ravel(A).
+        Assertions.assertArrayEquals(
+                (int[]) input.get1DJavaArray(ucar.ma2.DataType.UINT),
+                (int[]) encoded.get1DJavaArray(ucar.ma2.DataType.UINT));
+
+        // decode is the inverse of encode.
+        ucar.ma2.Array decoded = codec.decode(encoded);
+        Assertions.assertArrayEquals(inputShape, decoded.getShape());
+        assert MAMath.equals(input, decoded);
```
Comment (normanrz) [inline src/test/java/dev/zarr/zarrjava/ReshapeCodecTest.java]: The `assert` keyword should not be used. Replace with:
```suggestion
        Assertions.assertTrue(MAMath.equals(input, decoded));
```

File: src/test/java/dev/zarr/zarrjava/ReshapeCodecTest.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        int[] testData = new int[4 * 5 * 6 * 3];
+        Arrays.setAll(testData, p -> p);
+
+        StoreHandle storeHandle = new FilesystemStore(TESTOUTPUT).resolve("reshape_single_chunk");
+        ArrayMetadataBuilder builder = Array.metadataBuilder()
+                .withShape(4, 5, 6, 3)
+                .withDataType(DataType.UINT32)
+                .withChunkShape(4, 5, 6, 3)
+                .withFillValue(0)
+                .withCodecs(c -> c.withReshape(new Object[]{new int[]{0, 1}, new int[]{2}, 3}).withZstd());
```
Comment (normanrz) [inline src/test/java/dev/zarr/zarrjava/ReshapeCodecTest.java:178]: Once we landed #81, we should add a test that uses reshape + jpeg. Because reshape does not alter the byte-layout, this test doesn't do much. Still useful, though!

**Review by normanrz — APPROVED** [review]
Looks good, just a few minor notes.
